h5py/tests/test_dataset.py

```diff
@@ -165,6 +166,63 @@ def test_empty_create_via_Empty_class(self):
         self.f.create_dataset('foo', data=h5py.Empty(dtype='f'))
         self.assertTrue(is_empty_dataspace(self.f['foo'].id))
 
+    def test_create_incompatible_data(self):
+        data = self.f.create_dataset('dset', (3,))
+        with self.assertRaises(ValueError):
+            self.f.create_dataset('bar', shape=4, data=data)
```

Comment:
If you're just testing the difference in shape, can you make `data` a numpy array? It's slightly confusing to have it be another dataset, unless that's the point of the test.

Reply:
yes, I can.